src/utils/image.js

```diff
@@ -304,6 +307,95 @@ export class RawImage {
         return this._update(newData, this.width, this.height, 4);
     }
 
+    /**
+     * Get the pixel at the given coordinates.
+     * @param {number} x The x coordinate of the pixel.
+     * @param {number} y The y coordinate of the pixel.
+     * @returns {Promise<DataArray>} The pixel value. The length of the array will match the channels.
+     */
+    async getPixel(x, y) {
+        x = Math.max(Math.min(x, this.width), 0);
+        y = Math.max(Math.min(y, this.height), 0);
+
+        // Calculate base index, taking into account number of channels.
+        const baseIndex = ((y * this.width) + x) * this.channels;
+        return this.data.slice(baseIndex, baseIndex + this.channels);
+    }
+
+    /**
+     * Set the pixel at the given coordinates.
+     * @param {number} x The x coordinate of the pixel.
+     * @param {number} y The y coordinate of the pixel.
+     * @param {DataArray} value The pixel value. The length of the array should match the channels.
+     * @returns {Promise<RawImage>}
+     * @throws {Error} If the number of channels in the value does not match the number of channels in the image.
+     */
+    async setPixel(x, y, value) {
+        x = Math.max(Math.min(x, this.width), 0);
+        y = Math.max(Math.min(y, this.height), 0);
+
+        if (value.length !== this.channels) {
+            throw new Error(`Expected ${this.channels} values, got ${value.length}`);
+        }
+
+        // Calculate the starting point for that pixel, by first determining
+        // what row to start on, then adding the x offset. Finally, take into
+        // account the number of channels.
+        const baseIndex = ((y * this.width) + x) * this.channels;
+
+        // Iterate for each channel, assigning the value to the corresponding
+        // spot in `this.data`.
+        for (let channel = 0; channel < this.channels; channel++) {
+            const channelIndex = baseIndex + channel;
+            if (channelIndex >= this.data.length) {
+                throw new Error('Index out of bounds');
+            }
+            this.data[channelIndex] = value[channel];
+        }
+
+        return this;
+    }
```

Comment:
```suggestion
```

Since these functions aren't used in this PR, I suggest we remove them and (if needed) add them to a separate PR. Is there a specific reason for their inclusion?

Reply:
Good catch!  
Originally I was using both of these functions during the loop, but I removed them as the overhead was not worth it.